pkgs/games/katago/default.nix

```
@@ -67,25 +68,27 @@ stdenv.mkDerivation rec {
   ];
 
   cmakeFlags = [
-    "-DNO_GIT_REVISION=ON"
+    (lib.cmakeBool "NO_GIT_REVISION" true)
   ] ++ lib.optionals enableAVX2 [
-    "-DUSE_AVX2=ON"
+    (lib.cmakeBool "USE_AVX2" true)
   ] ++ lib.optionals (backend == "eigen") [
-    "-DUSE_BACKEND=EIGEN"
+    (lib.cmakeFeature "USE_BACKEND" "EIGEN")
```

As far as I remember, CMake is case-insensitive.

Nonetheless, you can write:

```suggestion
    (lib.cmakeFeature "USE_BACKEND" (lib.toUpper backend))
```

and remove that truckload of backend tests

---

oh man!  I guess that does make a lot of sense.  Thanks for the help cleaning this stuff up.